### ISSUE TITLE
Signature Recognition

### ISSUE THREAD

**Issue by basketballautomation** [AlexeyAB/darknet#3956]
I tried training a model on a database of signature images (so just 1 class) but have had no luck with it picking up the signatures from the test data. Conceptually, is there a reason this wouldn't work for signatures?

Thanks!

**Comment by AlexeyAB** [comment]
Can you show Loss & mAP chart of your training?
What mAP did you get on Training and Test data?

**Comment by basketballautomation** [comment]
For a bit of context, I ran the following first:

`Darknet_no_gpu.exe detector train dta/obj.data yolov3-tiny_obj.cfg yolov3-tiny.weights -dont_show`

Subsequently, I ran the following and received the screenshot as output. Considering the 0%, I'm wondering if I did something wrong earlier on.

`Darknet_no_gpu.exe detector map data/obj.data yolov3-tiny_obj.cfg yolov3-tiny.weights`

![1-Map Results](https://user-images.githubusercontent.com/55562166/65338029-5854ae00-db97-11e9-9768-73690ba3a342.PNG)

**Comment by AlexeyAB** [comment]
You should train several years without GPU `Darknet_no_gpu.exe `

**Comment by basketballautomation** [comment]
Sorry, what exactly do you mean by several years? Just increase my training set data size in terms of quantity?

**Comment by AlexeyAB** [comment]
How many images do you have and how many iterations did you train? 
Is it more than 4000?

**Comment by basketballautomation** [comment]
-I believe my training set consisted of ~700 or 800 images. What would recommend?
-And sorry if I should/know or remember this, but how do I know how many iterations I ran?

**Comment by AlexeyAB** [comment]
Read this about iterations number: https://github.com/AlexeyAB/darknet#when-should-i-stop-training
And train by using GPU.

**Comment by basketballautomation** [comment]
I am now training with GPU and have adjusted iterations. I am currently training (been running for ~10 hours) and my avg loss seems very high and the Rate of 0.0000 seems incorrect. Any further suggestions or anything else I can provide to help try and solve this issue?

![M_AP](https://user-images.githubusercontent.com/55562166/65645651-41aebc80-dfc6-11e9-8a3d-4dfea36120ec.png)

**Comment by AlexeyAB** [comment]
* You trained only 9 iterations. But you should train at least 4000 iterations: https://github.com/AlexeyAB/darknet#when-should-i-stop-training

* Your training time for 1 iteration is 3012 seconds, so you don't use GPU

* learning rate ~ 0 is normal for the first 1000 iterations due to: https://github.com/AlexeyAB/darknet/blob/05545b260784cb5ce9b27f85462c4a31137b684b/cfg/yolov3.cfg#L19

**Comment by basketballautomation** [comment]
Thank you for the quick response.

That all makes sense, except that I certainly installed it to use GPU (or at least intended to). Are there any typical errors seen in processing that wouldn't use GPU? I tried to follow all installation instructions perfectly and am using Darknet.exe, but must have made a mistake somewhere.

**Comment by basketballautomation** [comment]
I am trying to rebuild the solution to confirm that I have GPU working, but am now receiving this error/issue in Microsoft Visual Studio. Have you (or anyone) seen this issue before or know how to solve it? (Using Microsoft and CMAKE-GUI Installation method).
![MVS_Output](https://user-images.githubusercontent.com/55562166/65721693-8004b400-e078-11e9-8667-de27e5d72a31.png)
 Thanks!

**Comment by basketballautomation** [comment]
Hey Alexey - any reason that you can see per the screenshot why my avg loss is going to -nan here? or is best resolution to restart with most recently saved weights?

Thanks!
![image](https://user-images.githubusercontent.com/55562166/65990044-db5beb00-e458-11e9-959a-b982042eabcb.png)

**Comment by AlexeyAB** [comment]
Compile Darknet with GPU

**Comment by basketballautomation** [comment]
I was finally able to get GPU working through Google Colab (processing much faster), however, it stops usually after ~15 iterations as shown below. It doesn't show any errors, etc. Just stops as is. Any ideas? Thanks!!

![image](https://user-images.githubusercontent.com/55562166/66016557-20a60a00-e4a5-11e9-927f-7f9745c1c02f.png)

**Comment by AlexeyAB** [comment]
So does it freeze without exiting?

Look at the files `bad.list` and `bad_labels.list`

**Comment by basketballautomation** [comment]
Hi Alexey - thank you again for all of the help. The previous comment fixed my previous error.

Question on IoU threshold in detector.c --> do you have to rebuild for that change to be in effect? Or can I change that to say 0.6 and then run the "detector map" function and it will test for IoU threshold of 0.6?

Thanks.

**Comment by AlexeyAB** [comment]
@basketballautomation 
If you change c-file, you should rebuild Darknet.

**Comment by basketballautomation** [comment]
great - line 595 only (since line 725 is commented out)?
